app/services/promotions/general_price_service.rb

```diff
@@ -2,14 +2,19 @@ module Promotions
   class GeneralPriceService
     class << self
       def get_price_from_current_promotion(params)
+        # Interesting usage of the Promotion model here, what
+        # was your thought process/motivators to create it?
         promotion = Promotion.where('start_date <= ?', Time.current).where('end_date >= ?', Time.current).order(priority_level: :desc).first
+
         if promotion.present?
+          # Curious to know why did you come up with meta-programming for this solution.
```

Comment:
it could be use for another promotion too, so if change promotion, just add another promotion file and instance